sdk/dotnet/Pulumi/Deployment/Deployment.cs

```diff
@@ -93,27 +89,32 @@ private Deployment()
             _stackName = stack;
             _projectName = project;
 
-            _options = new Options(
-                queryMode: queryModeValue, parallel: parallelValue, pwd: pwd,
-                monitor: monitor, engine: engine, tracing: tracing);
+            Serilog.Log.Debug("Creating Deployment Engine and Monitor.");
+            this.RpcDispatcher = new GrpcDispatcher(engine, monitor);
+            Serilog.Log.Debug("Created Deployment Engine and Monitor.");
 
-            Serilog.Log.Debug("Creating Deployment Engine.");
-            this.Engine = new Engine.EngineClient(new Channel(engine, ChannelCredentials.Insecure));
-            Serilog.Log.Debug("Created Deployment Engine.");
-
-            Serilog.Log.Debug("Creating Deployment Monitor.");
-            this.Monitor = new ResourceMonitor.ResourceMonitorClient(new Channel(monitor, ChannelCredentials.Insecure));
-            Serilog.Log.Debug("Created Deployment Monitor.");
+            _runner = new Runner(this);
+            _logger = new Logger(this, this.RpcDispatcher);
+        }
 
+        /// <summary>
+        /// This constructor is called from <see cref="TestAsync{TStack}"/> with
+        /// a mocked monitor and dummy values for project and stack.
+        /// </summary>
+        private Deployment(IRpcDispatcher rpcDispatcher)
+        {
+            _isDryRun = true;
+            _stackName = "teststack";
+            _projectName = "testproject";
```

Comment:
Feels a little odd to set these to test names at this level. Also - these names seem different than in https://github.com/pulumi/pulumi/pull/3738.

Reply:
What would be the right level to set them?
(I'll change it to `stack` and `project`.)

Reply:
Changed to `stack` and `project`.